src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1217,10 +1217,10 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
                 "BCP200",
                 $"{BuildInvalidOciArtifactReferenceClause(aliasName, badRef)} The registry \"{badRegistry}\" exceeds the maximum length of {maxLength} characters.");
 
-            public ErrorDiagnostic ExpectedProviderSpecification() => new(
+            public ErrorDiagnostic ExpectedProviderSpecificationOrCompileTimeImportExpression() => new(
                 TextSpan,
                 "BCP201",
-                "Expected a provider specification string. Specify a valid provider of format \"<providerName>@<providerVersion>\".");
+                "Expected a provider specification string of format \"<providerName>@<providerVersion>\", an asterisk, or a left curly brace.");
```

Comment:
Why not just reference the characters? `.., a '*' character or a '{' character`? That would be more consistent with other parser errors.